packages/data/src/world/mq/ganon_castle_mq.yml

```diff
@@ -35,92 +39,256 @@
   locations:
-    "MQ Ganon Castle SR Fire Under Pillar": "has_tunic_goron && can_lift_gold"
+    "MQ Ganon Castle Forest Trial Key": "can_hookshot || can_boomerang || climb_anywhere"
+"Ganon Castle Forest Trial Wind Room Starting Side":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Forest Trial Entrance": "true"
+    "Ganon Castle Forest Trial Wind Room Ending Side": "true" #Can use wind to jump to the other side.
+  events:
+    GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE: "has_fire_arrows"
+    GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE: "can_hit_triggers_distance"
+  locations:
+    "MQ Ganon Castle Forest Trial First Chest": "event(GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE)"
+    "MQ Ganon Castle Forest Trial Second Chest": "event(GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE)"
+"Ganon Castle Forest Trial Wind Room Ending Side":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Forest Trial Wind Room Starting Side": "hookshot_anywhere || (has_hover_boots && climb_anywhere) || (can_longshot && event(GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE))"
+    "Ganon Castle Forest Trial Ending Room": "event(GANON_MQ_FOREST_SWITCH)"
+  events:
+    GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE: "has_fire"
+    GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE: "can_hit_triggers_distance"
+    GANON_MQ_FOREST_SWITCH: "can_play_time || can_play_elegy"
+  locations:
+    "MQ Ganon Castle Forest Trial Second Chest": "event(GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE)"
+"Ganon Castle Forest Trial Ending Room":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Forest Trial Wind Room Ending Side": "true"
+  events:
+    GANON_TRIAL_FOREST: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Forest End 1": "true"
+    "MQ Ganon Pot Forest End 2": "true"
+"Ganon Castle Fire Trial Entrance Ledge":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Main Room": "true"
+    "Ganon Castle Fire Trial Ending Ledge": "has_tunic_goron && ((can_longshot && silver_rupees_ganon_fire) || has_hover_boots || (hookshot_anywhere && event(GANON_MQ_FIRE_MONOLITH)))"
+  events:
+    GANON_MQ_FIRE_MONOLITH: "has_tunic_goron && can_lift_gold"
+  locations:
+    "MQ Ganon Castle SR Fire Under Pillar": "event(GANON_MQ_FIRE_MONOLITH)"
     "MQ Ganon Castle SR Fire Center-Left": "has_tunic_goron_strict"
     "MQ Ganon Castle SR Fire Back-Left": "has_tunic_goron_strict"
     "MQ Ganon Castle SR Fire High Above Lava": "has_tunic_goron"
     "MQ Ganon Castle SR Fire Front-Left": "has_tunic_goron_strict"
-    "MQ Ganon Pot Fire End 1": "has_tunic_goron_strict && (can_longshot || has_hover_boots) && silver_rupees_ganon_fire"
-    "MQ Ganon Pot Fire End 2": "has_tunic_goron_strict && (can_longshot || has_hover_boots) && silver_rupees_ganon_fire"
-"Ganon Castle Water":
+"Ganon Castle Fire Trial Ending Ledge":
   dungeon: Ganon
+  exits:
+    "Ganon Castle Fire Trial Entrance Ledge": "has_tunic_goron && (longshot_anywhere || (hookshot_anywhere && event(GANON_MQ_FIRE_MONOLITH)))"
```

Comment:
The monolith does not stay. it resets upon reloading the room. Also forgot to mention climb anywhere can do with strict.

Reply:
Reverted, and added climb anywhere